Homebrew install on M1 Mac broken due to UPX defect

When UPX is present, `make install` on an M1 Mac results in a binary that does not execute. This includes Homebrew installs since UPX is a formula dependency:

```
% /opt/homebrew/bin/unifi-poller
zsh: killed     /opt/homebrew/bin/unifi-poller
```

This appears to be due to [UPX issue #446](https://github.com/upx/upx/issues/446) which causes packed darwin/arm64 binaries to have a corrupted executable header.

The fix is to ensure that UPX is not installed before doing `make install`. The resulting executable works fine:

```
 % unpoller                                              
2021/08/29 15:51:01 [ERROR] finding file: stat /usr/local/etc/unifi-poller/up.conf: no such file or directory
```

Should we consider skipping the UPX dependency when executing on Mac arm64?

--- a/Makefile
+++ b/Makefile
@@ -12,11 +12,11 @@ MD2ROFF_BIN=github.com/davidnewhall/md2roff@v0.0.1
 RSRC_BIN=github.com/akavel/rsrc
 
 # If upx is available, use it to compress the binaries.
+UPXPATH=$(shell which upx)
+
 # Skip upx in Mac ARM environments: https://github.com/upx/upx/issues/446
 ifeq ($(shell uname -ps),Darwin arm)
 	UPXPATH=
-else
-	UPXPATH=$(shell which upx)
 endif
 
 # Travis CI passes the version in. Local builds get it from the current git tag.

--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,12 @@ MD2ROFF_BIN=github.com/davidnewhall/md2roff@v0.0.1
 RSRC_BIN=github.com/akavel/rsrc
 
 # If upx is available, use it to compress the binaries.
-UPXPATH=$(shell which upx)
+# Skip upx in Mac ARM environments: https://github.com/upx/upx/issues/446
+ifeq ($(shell uname -ps),Darwin arm)
+	UPXPATH=
+else
+	UPXPATH=$(shell which upx)
+endif
 
 # Travis CI passes the version in. Local builds get it from the current git tag.
 ifeq ($(VERSION),)

